tests/e2e/cjs/index.test.js

```diff
@@ -277,3 +287,53 @@ describe('ANTRegistry', async () => {
     assert(Array.isArray(affiliatedAnts.Controlled));
   });
 });
+
+describe('Signing', async () => {
+  let signers = [];
+  before(async () => {
+    const jwk = await arweave.wallets.generate();
+    signers = [new ArweaveSigner(jwk), createAoSigner(new ArweaveSigner(jwk))];
+  });
+
+  it('Should be able to sign on the IO contract with all ContractSigner types', async () => {
+    for (const signer of signers) {
+      const io = IO.init({ signer });
+      const res = await io
+        .transfer({
+          target: 'QGWqtJdLLgm2ehFWiiPzMaoFLD50CnGuzZIPEdoDRGQ',
+          qty: 1,
+        })
+        .catch((e) => e);
+      // if it is a WriteInteractionError, it means the transaction was signed contract threw an error
+      assert(res instanceof WriteInteractionError);
+    }
```

Comment:
we should not be sending any real messages - the tests should only be validating the `init` function that accepts the signer and the expected instance type is returned.